### PR TITLE
Handle sync errors

### LINKED ISSUE #791: Kinto Error HTTP 507 Insufficient Storage
On authentication using FxA, Kinto print in console an error 507

```
Error: HTTP 507; Error: HTTP 507 Insufficient Storage: Resource access is forbidden for this user (Collection maximum size exceeded (205369 > 204800 Bytes).)
Stack trace:
formatResponse@moz-extension://6aefee5b-5199-4743-b265-b47b7d8779ce/vendor/kinto.js:4145:21
_callee$@moz-extension://6aefee5b-5199-4743-b265-b47b7d8779ce/vendor/kinto.js:4097:50
r@moz-extension://6aefee5b-5199-4743-b265-b47b7d8779ce/vendor/kinto-http.js:7:204
[2]</</u/<@moz-extension://6aefee5b-5199-4743-b265-b47b7d8779ce/vendor/kinto-http.js:7:1390
[2]</</a/</t[e]@moz-extension://6aefee5b-5199-4743-b265-b47b7d8779ce/vendor/kinto-http.js:7:380
step@moz-extension://6aefee5b-5199-4743-b265-b47b7d8779ce/vendor/kinto.js:3958:183
[9]</_asyncToGenerator/</</step/<@moz-extension://6aefee5b-5199-4743-b265-b47b7d8779ce/vendor/kinto.js:3958:361
s@moz-extension://6aefee5b-5199-4743-b265-b47b7d8779ce/vendor/kinto-http.js:596:1731
[200]</notify/<@moz-extension://6aefee5b-5199-4743-b265-b47b7d8779ce/vendor/kinto-http.js:596:1870
s@moz-extension://6aefee5b-5199-4743-b265-b47b7d8779ce/vendor/kinto-http.js:193:343
  sync.js:265:7
```

Having lots of async multi-thread stuff going on, I had a couple of infinite loop populating my storage. On sync I receive 370 skipped entries having deleted: true.

Still trying to use it, seams like it keeps increasing with time.
```Error: HTTP 507; Error: HTTP 507 Insufficient Storage: Resource access is forbidden for this user (Collection maximum size exceeded (212530 > 204800 Bytes).)```

### PR DESCRIPTION
**Pull request by sebastienbarbier**
After authenticating, kinto might return some errors we should handle.

- [x] fix synced UI after kinto error. Should request reconnect.
- [x] implement a solution to unblock the user.
- [x] display used quota. #837
- [ ] display an error message to inform the user.

Fix #791 #801 #802 #827

### CODE REVIEW

**Comment by sebastienbarbier** [comment]
Waiting for @ryanfeeley to suggest an error panel, but did this temporary UI just to proove we can handle that issue properly 🎉

errormessage

**Comment by ryanfeeley** [comment]
@sebastienbarbier I could not have done it better myself!

Here is where we should put storage information.
image

I'm working on when/how we should warn them BEFORE IT’S TOO LATE.

File: src/sync.js [hunk 1]
```diff
@@ -104,6 +102,12 @@ class JWETransformer {
     // If we get such a record, flag it as deleted.
     if (decoded._status === 'deleted') {
       decoded.deleted = true;
+      // On decode, we flag notes with _status deleted but still on server.
+      // We automatically will request deletion for those.
+      // (This is due to singleNote overriding deleted state with udpated to server)
```
Comment (vladikoff) [inline src/sync.js]: typo `udpated`

File: src/sync.js [hunk 2]
```diff
@@ -104,6 +102,12 @@ class JWETransformer {
...
+      // Should be deleted when every user who tryed beta runned it once.
```
Comment (vladikoff) [inline src/sync.js]: typo `tryed`

File: src/sync.js [hunk 3]
```diff
@@ -104,6 +102,12 @@ class JWETransformer {
...
+      // (see meetrics deleteDeleted)
```
Comment (vladikoff) [inline src/sync.js]: typo `meetrics`

**Comment by sebastienbarbier** [comment]
I'll refactor and yes we should be able to merge soon.

**Comment by flodolo** [comment]
Is there a point of flagging me for review if these PRs get merged within hours, without giving me a chance to look at them? That's the second time it happens in a few hours.

**Comment by vladikoff** [comment]
> Is there a point of flagging me for review if these PRs get merged within hours, without giving me a chance to look at them? That's the second time it happens in a few hours.

Sorry, yeah good point. I Think we are gonna change the process on our end and not introduce strings in small batches, and do a separate commit for those before when preparing a release so you get a chance to review them (in a larger batch as well).

**Pull request merged**